Makefile

```diff
@@ -45,13 +58,29 @@ vet:
 # build targets
 #
 
-bin/%:
-	$(Q)echo "Building $@..."; \
-	$(GO_BUILD) -o $@ ./$(subst bin/,cmd/,$@)
+$(BINARIES): bin/%:
+	$(Q)echo "Building $@..."
+	$(Q)(cd cmd/$(*) && $(GO_BUILD) -o $(abspath $@) .)
+
+#
+# vendoring targets
+#
+.PHONY: mod-verify mod-tidy $(MOD_TIDY_TARGETS) $(MOD_VERIFY_TARGETS)
+
+MOD_TARGETS := $(CMDS) ..
```

Comment:
So do I read this correctly ?

This generates a target for each binary in `cmd` and an extra one `..`. Then `..` gets eventually expanded as `mod-tidy-..` (+ ditto for verify), so it runs the top-level `go mod tidy` via evaluating `$(abspath ./cmd/..)` ?

Does this work correctly ? I suspect that it might not in some cases. I mean, shouldn't we run `go mod tidy` *first* in top-level directory (since that is a go-level dependency of each command) and only then in the subdirectories so any dep-changes in the top-level go module are reflected in the commands' modules. I think now it could happen that the subdirs are not correctly updated. IOW, `..` should at least come first here ?

Also, (and I admit that I am often guilty of going too ballistic with `Makefiles`, so I don't know if I can really complain about this with a straight face) I think that how the top-dir/"`..`" gets tidied might be a bit too quirky. Maybe we could have something more explicit/straightforward with a smaller chance to bite us later in the arse, once we ourselves don't understand/remember any more how this was supposed to work.

Reply:
It might make sense to split out the top-level targets here to avoid assumptions like this not working as expected.

Let me clean it up a bit.

Reply:
I have updated this to be more explicit. Please take a look.